util/threadpool_imp.cc

```diff
@@ -227,14 +222,18 @@ void ThreadPoolImpl::Impl::BGThread(size_t thread_id) {
                      std::memory_order_relaxed);
 
     bool decrease_io_priority = (low_io_priority != low_io_priority_);
-    bool decrease_cpu_priority = (low_cpu_priority != low_cpu_priority_);
+    CpuPriority cpu_priority = cpu_priority_;
     lock.unlock();
 
 #ifdef OS_LINUX
-    if (decrease_cpu_priority) {
+    if (cpu_priority < current_cpu_priority) {
+      TEST_SYNC_POINT_CALLBACK("ThreadPoolImpl::BGThread::BeforeSetCpuPriority",
+                               &current_cpu_priority);
       // 0 means current thread.
-      port::SetCpuPriority(0, CpuPriority::kLow);
-      low_cpu_priority = true;
+      port::SetCpuPriority(0, cpu_priority);
```

Comment:
Since we've called a function in port, Do we still need to guard it with OS_LINUX?

Reply:
removed